src/main/kotlin/cacophony/controlflow/Variable.kt

```diff
@@ -0,0 +1,17 @@
+package cacophony.controlflow
+
+import cacophony.semantic.syntaxtree.Definition
+
+sealed class Variable
+
+class SourceVariable(
+    val definition: Definition,
+) : Variable()
+
+sealed class Register() : Variable()
+
+class VirtualRegister : Register()
```

Comment:
**Question to all:** Do we want to leave `VirtualRegister` outside of `Register` or maybe put it inside as `Register.Virtual`?